crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,7 +287,101 @@ impl Executor {
                     }
                 }
             }
+        };
+
+        match &reason {
+            Ok(reason) => {
+                info!(reason, "shutting down");
+            }
+            Err(reason) => {
+                error!(%reason, "shutting down");
+            }
+        };
+
+        let mut total_bundles_drained_successfully = 0;
+        let shutdown_logic = async {
+            info!("setting executor connection status to disconnected");
+            self.status
+                .send_modify(|status| status.is_connected = false);
+
+            info!("draining sequence actions from the executor receiver channel");
+            // drain the receiver channel
+            while let Ok(seq_action) = self.serialized_rollup_transactions.try_recv() {
```

Comment:
Shouldn't this channel be closed before trying to drain it?

Reply:
you are right, i missed that